src/dcqc/file.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import annotations
+
+import os
+import re
+from collections.abc import Collection, Mapping
+from copy import deepcopy
+from dataclasses import asdict, dataclass
+from pathlib import Path
+from typing import Any, Optional
+
+from dcqc.mixins import SerializableMixin
+from dcqc.utils import open_parent_fs
+
+
+@dataclass
+class FileType:
+    # Class attributes
+    # A type hint is omitted so this attribute isn't
+    # picked up by @dataclass as an instance attribute
+    _registry = dict()  # type: ignore
+
+    # Instance attributes
+    name: str
+    file_extensions: tuple[str, ...]
+
+    def __init__(self, name: str, file_extensions: Collection[str]):
+        self.name = name
+        self.file_extensions = tuple(file_extensions)
+        self.register_file_type(self)
+
+    @classmethod
+    def register_file_type(cls, self):
+        name = self.name.lower()
+        if name in cls._registry:
+            message = f"File type ({name}) is already registered ({self._registry})."
+            raise ValueError(message)
+        cls._registry[name] = self
+
+    @classmethod
+    def get_file_type(cls, file_type: str) -> FileType:
+        file_type = file_type.lower()
+        if file_type not in cls._registry:
+            types = list(cls._registry)
+            message = f"File type ({file_type}) not among available options ({types})."
+            raise ValueError(message)
+        return cls._registry[file_type]
+
+
+# Instantiated file types are automatically tracked by the FileType class
+FileType("*", ())  # To represent all file types
```

Comment:
I'm ok with this, but what are your thoughts on a configuration file that we can add and remove from to track these FileType classes? 

Perhaps add a TODO if you think thats a good idea?

Reply:
The idea of a "database" of file types (_e.g._ with a CSV file) also crossed my mind. I just didn't see any benefits for the time being. Can you think of any? 

While it would allow users to add their own file types, they wouldn't be able to take advantage of them unless they also created a suite. Also, they can already add new file types by instantiating the `FileType` class; they're automatically tracked wherever they are created. 

I did add a TODO to track the idea in case something changes and makes the change worthwhile.

Reply:
Ah, thats what I was missing - users need to create a suite to take advantage of their own filetypes.  Not sure how feasible this is, but one use case I can think of for a "database" of filetypes is if we can encapsulate the information required for both `FileType` and `Suite`.

Reply:
Yup, I have [another TODO](https://github.com/Sage-Bionetworks-Workflows/py-dcqc/blob/0ddb5837795b7ece8afd90246bb4ec4034e1a0b0/src/dcqc/suites/suites.py#L6-L7) that would have the file types also include the associated with the suites. We would have to figure out a way to ensure that a dependency isn't introduced from file types to suites, which would break the layered architecture.